File: java/Stack/easy/make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        if (isGood(s)<0)return s;
        else{
            while(isGood(s)>=0){
                if(s.length()==2){return "";}
                int k=isGood(s);
                s=s.substring(0,k)+s.substring(k+2);
            }
            return s;
        }
    }
    public int isGood(String s){
       
        for(int i=0;i<=s.length()-2;i++){
            int a=(int) s.charAt(i);
            int b=(int) s.charAt(i+1);
            if (a-32==b||b-32==a){
                return i;
            }
        }
        return -1;
    }
}

File: cpp/Stack/easy/make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char> st;
        for(auto i : s) {
            if(st.empty()) {
                st.push(i);
            } else {
                if(i >= 'A' && i <= 'Z' && st.top() >= 'a' && st.top() <= 'z' && tolower(i) == st.top()) {
                    st.pop();
                } else if(i >= 'a' && i <= 'z' && st.top() >= 'A' && st.top() <= 'Z' && tolower(st.top()) == i) {
                    st.pop();
                } else {
                    st.push(i);
                }
            }
        }
        string ans = "";
        while(!st.empty()) {
            ans += (st.top());
            st.pop();
        }
        return string(ans.rbegin(), ans.rend());
    }
};
